clang/lib/CodeGen/BackendUtil.cpp

```diff
@@ -1052,6 +1053,9 @@ void EmitAssemblyHelper::RunOptimizationPipeline(
       // Allocate static local memory in SYCL kernel scope for each allocation
       // call.
       MPM.addPass(SYCLLowerWGLocalMemoryPass());
+
+      // Process properties and annotations
+      MPM.addPass(CompileTimePropertiesPass());
```

Comment:
I'm wondering if we should somehow test that clang output now have the corresponding transformation applied?

Reply:
What do you suggest?

Reply:
There are couple of ways to do so:

Write an integration test which invokes clang FE stage + clang CodeGen stage and inspect resulting LLVM IR with `FileCheck` to ensure that we see transformations don by the pass. An example would be (or any other test in `sycl/test/check_device_code`): https://github.com/intel/llvm/blob/3d8a136c0b1b230de998c7b34d716ececa868187/sycl/test/check_device_code/buffer_location_codegen.cpp#L1-L4

Alternative more light-weight approach is to print list of passes in a pipeline to check that the pass is there. For that you can modify existing test: https://github.com/intel/llvm/blob/3d8a136c0b1b230de998c7b34d716ececa868187/clang/test/CodeGenSYCL/kernel-early-optimization-pipeline.cpp#L1-L11

My preference would be to simply check for the pass presence in optimization pipeline: perhaps we should generalize  existing (or create another) test to check for all SYCL-specific passes and not only whether or not optimization passes were enabled/disabled.

Reply:
I went with the latter approach in the most recent (hopefully last) commit. Hope this finishes things off.